tasks/system_probe.py

```diff
@@ -1705,7 +1705,7 @@ def generate_minimized_btfs(ctx, source_dir, output_dir, bpf_programs):
         nw.rule(name="minimize_btf", command="bpftool gen min_core_btf $in $out $input_bpf_programs")
         nw.rule(
             name="compress_minimized_btf",
-            command="tar --mtime=@0 -cJf $out -C $tar_working_directory $rel_in && rm $in",
+            command=f"tar cf $out --mtime=@0 -I 'xz -8 -T{os.environ.get('KUBERNETES_CPU_REQUEST', 1)}' -C $tar_working_directory $rel_in && rm $in",
```

Comment:
For `T{os.environ.get('KUBERNETES_CPU_REQUEST', 1)}`, I just found that:
> [...] a special value **0** makes **xz** use as many threads as there are CPU cores on the system.

Reply:
I think we want to avoid relying on things like `nproc` in case we end up using the physical CPU count rather than the number of CPUs available to the runner, but I agree that it would be much cleaner/easier